apps/src/code-studio/initApp/project.js

```diff
@@ -1042,15 +1042,20 @@ var projects = (module.exports = {
         const html = this.sourceHandler.getLevelHtml();
         const makerAPIsEnabled = this.sourceHandler.getMakerAPIsEnabled();
         const selectedSong = this.sourceHandler.getSelectedSong();
-        const libraries = this.sourceHandler.getLevelLibraries();
-        callback({
+        const libraries =
+          this.sourceHandler.getLevelLibraries &&
+          this.sourceHandler.getLevelLibraries();
+        var sourceAndHtml = {
           source,
           html,
           animations,
           makerAPIsEnabled,
-          selectedSong,
-          libraries
-        });
+          selectedSong
+        };
+        if (libraries) {
```

Comment:
likewise, is there a specific reason we want to avoid setting libraries here if it's undefined?

Reply:
(Answer to both questions), this is part of a prototype feature we are making for a pilot. The plan is to revert it when the pilot is finished and redesign it with what we learned from the pilot. While the pilot is running, however, I don't want to leave extra state/data sitting in projects that aren't participating in the pilot.